protos/core.proto

```diff
@@ -424,4 +427,16 @@ message TableMetadata {
   Target target = 1;
   string type = 2; // "table" or "view"
   repeated Field fields = 3;
+  int64 last_updated_millis = 4;
+}
+
+message PersistedTableMetadata {
+  Target target = 1;
+  message CacheKey{
```

Comment:
i think i change my mind back on this, sorry!

these two fields aren't really the cache key. the cache key really is just `definition_hash`. `last_updated_millis` is only used for computing whether a table's *dependencies* have changed; `last_updated_millis` for a table `A` is never used in the context of determining whether the cache for `A` is broken or not.

i think this was right the first time, just having these fields part of the outer message.

please put comments on the fields though, it's tricky to know precisely what they mean!